<launch>
    <!-- 
      1. Carica il file di configurazione YAML sul ROS Parameter Server.
         In questo modo il tuo nodo C++ potrà leggerlo facilmente.
    -->
    <rosparam command="load" file="$(find multi_robot_simulator)/config/sim_config.yaml" />

    <!-- 
      2. Lancia il tuo nodo C++ del simulatore.
         'pkg' è il nome del tuo pacchetto.
         'type' è il nome dell'eseguibile che hai definito in CMakeLists.txt.
         'name' è il nome che questo nodo avrà quando è in esecuzione.
         'output="screen"' mostra i messaggi (es. ROS_INFO) nel terminale.
    -->
    <node pkg="multi_robot_simulator" type="simulator_node" name="multi_robot_simulator_node" output="screen" />

    <!-- 
      3. Lancia RViz, lo strumento di visualizzazione di ROS.
         Opzionale ma FONDAMENTALE per vedere cosa succede.
         L'argomento '-d' carica una configurazione di RViz per non doverla rifare ogni volta.
         La prima volta lo configureremo e poi salveremo il file.
    -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find multi_robot_simulator)/config/simulation.rviz" required="true" />

</launch>